docs/how_to/upgrade-tf-controller.md

```diff
@@ -0,0 +1,22 @@
+# Upgrading TF-Controller
+
+Please follow these steps to upgrade TF-Controller:
+
+1. Read the latest release changelogs.
+2. Check your API versions.
+3. To make sure you don't get new state changes, suspend Terraform resources (`tfctl suspend --all`) to minimize the impact on live systems.
+4. Back up Terraform tfstates to avoid losing data. If you're using the default backend with secrets in Kubernetes, use your backup toolset (i.e., Velero) to back up the state data.
+5. Upgrade Flux first, following [the Flux documentation](https://fluxcd.io/flux/installation/upgrade/).
+6. To prevent unintentional resource deletions, set the `spec.destroyResourcesOnDeletion` flag to `False` for critical or production systems.
```

Comment:
```suggestion
6. To prevent unintentional resource deletions, set the `spec.destroyResourcesOnDeletion` flag to `false` for critical or production systems (the default value is `false`)
```